mithril-aggregator/src/services/snapshotter/compressed_archive_snapshotter.rs

```diff
@@ -548,6 +556,27 @@ mod tests {
             );
         }
 
+        #[tokio::test]
+        async fn getting_files_to_include_fails_when_no_ledger_file_found() {
+            let test_dir = temp_dir_create!();
+            let cardano_db = DummyCardanoDbBuilder::new(current_function!())
+                .with_immutables(&[1, 2])
+                .build();
+            let snapshotter =
+                snapshotter_for_test(&test_dir, cardano_db.get_dir(), CompressionAlgorithm::Gzip);
+            let ancillary_snapshot_dir = test_dir.join("ancillary_snapshot");
+            fs::create_dir(&ancillary_snapshot_dir).unwrap();
+
+            let err = snapshotter
+                .get_files_and_directories_for_ancillary_snapshot(1, &ancillary_snapshot_dir)
+                .await
+                .expect_err("Should fail if no ledger file found");
+
+            assert!(err
+                .to_string()
+                .contains("No ledger state snapshot found in the ledger directory"));
```

Comment:
We should probably either test that we have an error that downcasts to a specific error (created with `thiserror`) or just test that we have an error and not check the error message IMO.